### PR TITLE
notes for chapter17

### PR DESCRIPTION
**Pull request by Fgazzelloni**

### CODE REVIEW

**Review by jonthegeek — CHANGES_REQUESTED** [review]
A few things:
* We're trying to co-create a single set of notes that can be reused by future clubs. Can you merge the notes into the existing file for the corresponding chapters? If not, please put them in the `other` folder to be sorted out by later groups.
* The build failed. The code chunk where you create `bean_res` has `eval = FALSE`, but then you reference `bean_res` later. You need to define it in an evaluated chunk if you want to use it.
* Please name your `{r}` code chunks. That makes it *much* easier to debug (and someone in a future cohort might be doing the debugging, so they might not have any idea where to start). The names have to be unique (throughout the whole book), so something like `17-define-beans-res` is better than simply `define-variables`, for example.

Thanks for actively participating!

**Comment by Fgazzelloni** [comment]
Hello Jon,
hope now is fine. I have deleted the previous commit files, and directly proposed few major additions to 17-dimensionality_reduction.Rmd .
You can delete this pull request (I don't know how to do it) and check if you can see the changes I have made directly from github into 17-dimensionality_reduction.Rmd .

**Comment by jonthegeek** [comment]
We don't have to delete this PR, I'll review the latest changes you made as part of this PR and hopefully be able to accept soon! There's at least one edit left because the book still didn't build properly, but that will PROBABLY be more straightforward.

Thanks for understanding! We're going to make something great together!

I'll review this... maybe later today, but more likely it'll have to wait 'til Saturday.

**Comment by jonthegeek** [comment]
This is the error that is currently preventing the book from building:
```
Error in knitr::include_graphics(here::here("images/17-workflow.png")) : 
  Cannot find the file(s): "/Users/runner/work/bookclub-tmwr/bookclub-tmwr/images/17-workflow.png"
```

File: 17-dimensionality_reduction.Rmd [hunk 1]
```diff
@@ -10,132 +11,567 @@
...
+
+
+It is a technique which is very similar to PCA.
+
+PCA is a part of PLS, in the meaning that in addition to PCA, PLS makes use of the **outcome**. It practically makes the components considering the outcome values.
+
+
+
+```{r echo=FALSE, fig.align='center', fig.cap='[credits: PARTIAL LEAST-SQUARES REGRESSION: A TUTORIAL](http://home.mit.bme.hu/~horvath/IDA/1-s2.0-0003267086800289-main.pdf)', out.width='100%'}
+knitr::include_graphics(here::here('images/pls.png'))
```
Comment (jonthegeek) [inline 17-dimensionality_reduction.Rmd:269]:
```suggestion
knitr::include_graphics("images/pls.png")
```
In this context, it looks like `here::here` just confuses things. The relative path should do it. I'm going to try to accept these changes if I can so we can hopefully make this work.

File: 17-dimensionality_reduction.Rmd [hunk 2]
```diff
@@ -10,132 +11,567 @@
...
+
+- probabilities
+- normalizing (as the prob are not real prob as they can be greater than 1) it makes a proportion of the first results
+- repeating first steps
+- and comparing results 
+
+PLS algo:
+
+```{r echo=FALSE, fig.align='center', fig.cap='[credits: PARTIAL LEAST-SQUARES REGRESSION: A TUTORIAL](http://home.mit.bme.hu/~horvath/IDA/1-s2.0-0003267086800289-main.pdf)', out.width='100%'}
+knitr::include_graphics(here::here('images/pls_algo.png'))
```
Comment (jonthegeek) [inline 17-dimensionality_reduction.Rmd:326]:
```suggestion
knitr::include_graphics("images/pls_algo.png")
```

File: 17-dimensionality_reduction.Rmd [hunk 3]
```diff
@@ -145,20 +581,81 @@ bean_res <-
...
+# load(file = "data/bean_res.rda")
+# data(bean_res)
+```
+
+
+This is the first part of the workflow:
+
+
+```{r workflow_image,echo=FALSE, fig.align='center', fig.cap='bean_res: workflow', out.width='100%'}
+knitr::include_graphics(here::here('images/17-workflow.png'))
```
Comment (jonthegeek) [inline 17-dimensionality_reduction.Rmd:610]:
```suggestion
knitr::include_graphics("images/17-workflow.png")
```

**Review by jonthegeek — CHANGES_REQUESTED** [review]
Need to update references to some images. I'll accept these if possible to see if that fixes the build.

**Review by jonthegeek — CHANGES_REQUESTED** [review]
Ok, `here::here` wasn't the issue (thankfully). This file appears to be missing, please add it or remove the reference:

```Error in knitr::include_graphics("images/17-workflow.png") : 
  Cannot find the file(s): "images/17-workflow.png"
```

That said I haven't actually reviewed the changes yet, I'm just trying to get it to work so I can review it in context.

**Review by jonthegeek — CHANGES_REQUESTED** [review]
Please remove the doMC requirement and calls. That package isn't available for Windows, which makes this uneditable on a Windows machine. If we actually need it to compile the book, then we've definitely done something wrong.

One way to get around this issue is to make any code blocks that really need it `eval = FALSE`, and then show their output from RDS. But it really shouldn't be necessary to use this package in the notes of the club, because that makes the club inaccessible to Windows users.

Also please add titles for all `{r} blocks in the code. It makes debugging much easier.

File: 17-dimensionality_reduction.Rmd [hunk 4]
```diff
@@ -586,21 +634,15 @@ bean_res <-
...
 
-We do not actually read this back in these notes as the `.rda` file is too big for loading it.
 
-```{r file_read,eval=FALSE, include=T}
-# load(file = "data/bean_res.rda")
-# data(bean_res)
-```
+We do not actually read this back in these notes as the `.rda` file is too big for loading it, but you can use these commands on your machine:
+
+    bean_res <- readRDS(file = "data/bean_res.rda")
```
Comment (jonthegeek) [inline 17-dimensionality_reduction.Rmd:644]:
```suggestion
    bean_res <- readRDS(file = "data/bean_res.RDS")
```

File: 17-dimensionality_reduction.Rmd [hunk 5]
```diff
@@ -586,21 +634,15 @@ bean_res <-
...
 
 Then read it back, to have it again as a variable in the environment, and to be able to impute further work on it. In case of any crash of the system such as pausing times, you can always reload the models information without re-run the workflow which is time consuming.
 
-We do not actually read this back in these notes as the `.rda` file is too big for loading it.
 
-```{r file_read,eval=FALSE, include=T}
-# load(file = "data/bean_res.rda")
-# data(bean_res)
-```
+We do not actually read this back in these notes as the `.rda` file is too big for loading it, but you can use these commands on your machine:
```
Comment (jonthegeek) [inline 17-dimensionality_reduction.Rmd:642]:
```suggestion
We do not actually read this back in these notes as the `.rds` file is too big for loading it, but you can use these commands on your machine:
```

File: DESCRIPTION [hunk 6]
```diff
@@ -14,14 +14,17 @@ Imports:
...
     bestNormalize,
     bookdown,
+    broom.mixed,
     corrplot,
     corrr,
     DALEX,
     DALEXtra,
     details,
+    DiagrammeR,
+    doMC,
```
Comment (jonthegeek) [inline DESCRIPTION:24]: Please remove the doMC Import. If this package is used in your notes, turn off evaluation of those parts. doMC does not work on Windows, and the bookclubs are meant to be usable on all machines.

**Review by jonthegeek — CHANGES_REQUESTED** [review]
I'd like to get this merged in. Do you think you can make these changes and get it working?

**Comment by jonthegeek** [comment]
I'm going to go ahead and close this PR. When you get to this chapter with Cohort 4, we can sort out any relevant changes!

**Pull request closed without merging**